image/base/slim.Dockerfile

```diff
@@ -59,47 +58,30 @@ RUN cd / &&\
     npm install -g terser &&\
     npm install -g uglify-js
 
+ADD install-imagemagick /tmp/install-imagemagick
+RUN /tmp/install-imagemagick
+
+ADD install-jemalloc /tmp/install-jemalloc
+RUN /tmp/install-jemalloc
+
 ADD install-nginx /tmp/install-nginx
 RUN /tmp/install-nginx
 
-RUN apt -y install advancecomp jhead jpegoptim libjpeg-turbo-progs optipng
-
-RUN mkdir /oxipng-install && cd /oxipng-install &&\
-      wget https://github.com/shssoichiro/oxipng/releases/download/v5.0.1/oxipng-5.0.1-x86_64-unknown-linux-musl.tar.gz &&\
-      tar -xzf oxipng-5.0.1-x86_64-unknown-linux-musl.tar.gz && cd oxipng-5.0.1-x86_64-unknown-linux-musl &&\
-      cp oxipng /usr/local/bin &&\
-      cd / && rm -rf /oxipng-install
+ADD install-oxipng /tmp/install-oxipng
+RUN /tmp/install-oxipng
 
-RUN mkdir /jemalloc-stable && cd /jemalloc-stable &&\
-      wget https://github.com/jemalloc/jemalloc/releases/download/3.6.0/jemalloc-3.6.0.tar.bz2 &&\
-      tar -xjf jemalloc-3.6.0.tar.bz2 && cd jemalloc-3.6.0 && ./configure --prefix=/usr && make && make install &&\
-      cd / && rm -rf /jemalloc-stable
+ADD install-redis /tmp/install-redis
+RUN /tmp/install-redis
 
-RUN mkdir /jemalloc-new && cd /jemalloc-new &&\
-      wget https://github.com/jemalloc/jemalloc/releases/download/5.2.1/jemalloc-5.2.1.tar.bz2 &&\
-      tar -xjf jemalloc-5.2.1.tar.bz2 && cd jemalloc-5.2.1 && ./configure --prefix=/usr --with-install-suffix=5.2.1 && make build_lib && make install_lib &&\
-      cd / && rm -rf /jemalloc-new
+ADD install-ruby /tmp/install-ruby
+RUN /tmp/install-ruby
 
 RUN echo 'gem: --no-document' >> /usr/local/etc/gemrc &&\
-    mkdir /src && git -C /src clone https://github.com/rbenv/ruby-build.git &&\
-    cd /src/ruby-build && ./install.sh &&\
-    cd / && rm -rf /src/ruby-build && (ruby-build 2.7.5 /usr/local)
-
-RUN gem update --system
+    gem update --system
 
 RUN gem install bundler pups --force &&\
     mkdir -p /pups/bin/ &&\
-    ln -s /usr/local/bin/pups /pups/bin/pups &&\
-    rm -rf /usr/local/share/ri/2.7.5/system
-
-ADD install-redis /tmp/install-redis
-RUN /tmp/install-redis
-
-ADD install-imagemagick /tmp/install-imagemagick
-RUN /tmp/install-imagemagick
-
-# Validate install
-RUN ruby -Eutf-8 -e "v = \`convert -version\`; %w{png tiff jpeg freetype heic}.each { |f| unless v.include?(f); STDERR.puts('no ' + f +  ' support in imagemagick'); exit(-1); end }"
```

Comment:
Just to point out: this PR keeps ImageMagick install validation, but no longer has an explicit step of validating ruby installation. That line kinda did both, but I think we're fine.

Reply:
Ah I didn't consider that when I replace it. The `gem` steps immediately below the ruby install should also do a rudimentary validation of the ruby install.

I just dug a bit further on my CI runs here and noticed in the base `test_build` step:
```
 gem install net-http-persistent -v 4.0.1', require_name: 'net/http/persistent -i /var/www/discourse/plugins/discourse-algolia/gems/2.7.5 --no-document --ignore-dependencies --no-user-install
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["require_name: net/http/persistent"]
```

It's been like that for months as seen here https://github.com/discourse/discourse-algolia/blame/c57118ba425fecf9da33d77fb98e511b5fd88f98/plugin.rb#L10

It doesn't seem to be breaking the processes here for the base image (specs etc) but it's a bit odd and seems worth mentioning. I'm not sure exactly what it's trying to accomplish there.

Reply:
Fixed in https://github.com/discourse/discourse/pull/16192 😃